### PR TITLE
[python] open telemetry db integrations

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
## Description

New tests based on the automatic instrumentation with the Open Telemetry library for Python weblog application.
We launch tests that check the span generated for Open Telemetry library when we use different DB connections.
The validations are made on the traces sent from the Agent to the backend.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: utils/build/docker/python_otel/flask-poc-otel.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+COPY binaries* /binaries/
+RUN echo "1.0.0" > SYSTEM_TESTS_LIBRARY_VERSION
+RUN echo "1.0.0" > SYSTEM_TESTS_LIBDDWAF_VERSION
+RUN echo "1.0.0" > SYSTEM_TESTS_APPSEC_EVENT_RULES_VERSION
+
+COPY utils/build/docker/python/flask /app
+COPY utils/build/docker/python_otel/flask-poc-otel/app.py /app
+COPY utils/build/docker/python_otel/flask-poc-otel/app.sh /app
+
+#TODO RMM: Change docker flask-poc base to fix psycopg2 ( psycopg2-binary is not supported by open telemetry)
```
Comment (robertomonteromiguel) [inline utils/build/docker/python_otel/flask-poc-otel.Dockerfile:14]: I prefer to add these lines to flask-poc-base on a separate PR. I have to check that the change on docker base doesn't arise any issue on other tests

File: utils/build/docker/python_otel/flask-poc-otel/app.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+def hello_world():
+    return "Hello, World!\\n"
+
+
+@app.route("/db", methods=["GET", "POST", "OPTIONS"])
+def db():
+    service = flask_request.args.get("service")
+    operation = flask_request.args.get("operation")
+
+    print("REQUEST RECEIVED!")
```
Comment (mabdinur) [inline utils/build/docker/python_otel/flask-poc-otel/app.py:29]: nit: Are we running the flask app in debug_mode? If so we should already get debug logs when a request is received.

We  should either remove this print statement or make it more useful. Maybe add the endpoint and the service/operation name that was received.

File: utils/build/docker/python_otel/flask-poc-otel/app.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+opentelemetry-instrument gunicorn -w 1 -b 0.0.0.0:7777 --access-logfile - app:app -k gevent
```
Comment (mabdinur) [inline utils/build/docker/python_otel/flask-poc-otel/app.sh:3]: Just curious why are we using gunicorn? Can we just use the default flask development server.

**Review by mabdinur — APPROVED** [review]

File: utils/build/docker/python_otel/flask-poc-otel/app.py (hunk 2 above)
Comment (robertomonteromiguel) [reply utils/build/docker/python_otel/flask-poc-otel/app.py]: I changed the trace

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -593,10 +593,53 @@ jobs:
         DD_API_KEY_3: ${{ secrets.DD_API_KEY_3 }}
         DD_APP_KEY_3: ${{ secrets.DD_APP_KEY_3 }}
 
+    - name: Compress logs
```
Comment (cbeauchesne) [inline .github/workflows/ci.yml:596]: IIUC, what we really test here is the ability of the agent to handle open telemetry trace, right ?

**Review by cbeauchesne — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (robertomonteromiguel) [reply .github/workflows/ci.yml:596]: They are testing different "Open Telemetry scenarios".
There is a good documentation:

- https://datadoghq.atlassian.net/wiki/spaces/OTEL/pages/3012035497/OpenTelemetry+E2E+System+Tests
- https://docs.google.com/document/d/1uy73XwAc6QrcmfOFYLCTpOSgYZnGKOXlP9cnrWTIllo/edit#heading=h.6beuftc20yw3

When I'm done with the Open Telemetry and DB Integrations PRs, I'm thinking of documenting what I'm doing and taking the existing documentation about system-tests and Open Telemetry and bringing it into our confluence/system-tests

**Pull request merged**